ray-operator/controllers/ray/utils/util_test.go

```diff
@@ -840,3 +948,257 @@ func TestIsGCSFaultToleranceEnabled(t *testing.T) {
+			cluster: createRayClusterTemplate(headStuct, []struct {
+				replicas    *int32
+				minReplicas *int32
+				suspend     *bool
+				cpu         string
+				memory      string
+				numOfHosts  int32
+			}{
+				{
+					numOfHosts:  2,
+					replicas:    ptr.To[int32](3),
+					minReplicas: ptr.To[int32](0),
+					cpu:         "2",
+					memory:      "200Mi",
+					suspend:     ptr.To(true),
+				},
+			}),
+			expected: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("1"),
+				corev1.ResourceMemory: resource.MustParse("100Mi"),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := CalculateDesiredResources(tt.cluster)
+			assert.Equal(t, tt.expected.Cpu().String(), result.Cpu().String())
+			assert.Equal(t, tt.expected.Memory().String(), result.Memory().String())
+		})
+	}
+}
+
+func TestCalculateMinResources(t *testing.T) {
+	headStruct := struct {
+		cpu    string
+		memory string
+	}{
+		cpu:    "1",
+		memory: "100Mi",
+	}
+
+	tests := []struct {
+		cluster  *rayv1.RayCluster
+		expected corev1.ResourceList
+		name     string
+	}{
+		{
+			name:    "Single head pod with no worker groups",
+			cluster: createRayClusterTemplate(headStruct, nil),
+			expected: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("1"),
+				corev1.ResourceMemory: resource.MustParse("100Mi"),
+			},
+		},
+		{
+			name: "Head pod with one worker group",
+			cluster: createRayClusterTemplate(headStruct, []struct {
+				replicas    *int32
+				minReplicas *int32
+				suspend     *bool
+				cpu         string
+				memory      string
+				numOfHosts  int32
+			}{
+				{
+					numOfHosts:  2,
+					replicas:    nil,
+					minReplicas: ptr.To[int32](3),
+					cpu:         "1",
+					memory:      "200Mi",
+					suspend:     nil,
+				},
+			}),
+			expected: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("7"),
+				corev1.ResourceMemory: resource.MustParse("1300Mi"),
+			},
+		},
+		{
+			name: "Head pod with suspended worker group",
+			cluster: createRayClusterTemplate(headStruct, []struct {
+				replicas    *int32
+				minReplicas *int32
+				suspend     *bool
+				cpu         string
+				memory      string
+				numOfHosts  int32
+			}{
+				{
+					numOfHosts:  2,
+					replicas:    nil,
+					minReplicas: ptr.To[int32](3),
+					cpu:         "1",
+					memory:      "200Mi",
+					suspend:     ptr.To(true),
+				},
+			}),
+			expected: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("7"),
+				corev1.ResourceMemory: resource.MustParse("1300Mi"),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := CalculateMinResources(tt.cluster)
+			assert.Equal(t, tt.expected.Cpu().String(), result.Cpu().String())
+			assert.Equal(t, tt.expected.Memory().String(), result.Memory().String())
+		})
+	}
+}
+
+func TestCalculateResourcesDoesNotMutateCluster(t *testing.T) {
```

Comment:
don't need to have a separate test. Can we add the logic to `TestCalculateDesiredResources` and `TestCalculateMinResources` instead?